programs/fund-sender/src/utils/state.rs

```diff
@@ -89,3 +93,25 @@ pub struct SendFund<'info> {
     pub destination_account: UncheckedAccount<'info>,
     pub system_program: Program<'info, System>,
 }
+
+#[derive(Accounts)]
+pub struct StoreCertificates<'info> {
+    // to send the received retired climate token to a hold account
+    #[account(mut)]
+    pub payer: Signer<'info>,
+    pub state: Account<'info, State>,
+    #[account(
+        mut,
+        seeds = [OUTPUT_YIELD_ACCOUNT, &state.destination_seed, state.key().as_ref()],
+        bump = state.output_yield_account_bump,
+    )]
+    /// CHECK: Must be correctly derived from the state
+    pub output_yield_account: UncheckedAccount<'info>,
+    #[account(mut)]
+    /// CHECK: Must be correctly derived from the state
+    pub output_yield_token_account: Account<'info, TokenAccount>,
+    #[account(mut)]
+    /// CHECK: Must be correct destination account (check is done in instruction)
```

Comment:
You can add a check here to ensure that the owner is correct - something like:

```suggestion
#[account(token::authority = state.certificate_vault  @ ErrorCode::IncorrectHoldAccount)]
    pub token_account: Account<'info, TokenAccount>,
```

and then remove the equivalent check in the instruction.

Reply:
I have done it slightly differently since somehow it was complaining about the syntax when I use the snippet above directly, is this also ok?

>     #[account(
>         mut,
>         constraint = output_yield_token_account.owner == output_yield_account.key() @ ErrorCode::IncorrectTokenAccountOwner,
>     )]
>     ///  A token account owned by the outputYieldAccount
>     pub output_yield_token_account: Account<'info, TokenAccount>,